examples/dashboard-operator/pkg/controller/dashboard/dashboard_controller.go

```diff
@@ -71,8 +71,9 @@ func newReconciler(mgr manager.Manager) (r *ReconcileDashboard, srcLabels declar
 		declarative.WithStatus(status.NewBasic(mgr.GetClient())),
 		declarative.WithPreserveNamespace(),
 		declarative.WithApplyPrune(),
-		declarative.WithManagedApplication(srcLabels),
 		declarative.WithObjectTransform(addon.TransformApplicationFromStatus),
+		declarative.WithManagedApplication(srcLabels),
+		declarative.WithObjectTransform(addon.ApplyPatches),
```

Comment:
It is going to be weird if the field is there, but the patches don't actually do anything.  Maybe this should be the default?

Reply:
This is necessary today because the addons package defines the types, so the declarative package can't extract the patch out. We _could_ workaround this with reflection.

I think what we really need is an AddonsReconciler that applies all (or most?) of these transforms.